tests/test-extensions/test-localisation-properties-extension/src/main/java/org/mule/test/crafted/localisation/properties/extension/LocalisationConfigurationPropertiesProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.crafted.localisation.properties.extension;
+
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import org.mule.runtime.properties.api.ConfigurationProperty;
+import org.mule.runtime.properties.api.DefaultConfigurationPropertiesProvider;
+import org.mule.runtime.properties.api.ResourceProvider;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Optional;
+
+public class LocalisationConfigurationPropertiesProvider extends DefaultConfigurationPropertiesProvider {
+
+  private final static String LOCALE_PREFIX = "locale::";
+
+  public LocalisationConfigurationPropertiesProvider(ResourceProvider resourceProvider, String file) {
+    super(file, resourceProvider);
+  }
+
+  /**
+   * Returns the formatted given key according to the format specified on the configuration properties file
+   *
+   * @param configurationAttributeKey
+   * @return an Optional with the formatted value of the given key or {@link Optional#empty()} otherwise.
+   */
+  @Override
+  public Optional<ConfigurationProperty> provide(String configurationAttributeKey) {
+    if (configurationAttributeKey.startsWith(LOCALE_PREFIX)) {
+      String effectiveKey = configurationAttributeKey.substring(LOCALE_PREFIX.length());
+      ConfigurationProperty property = super.provide("language.pattern").get();
+      return of(new ConfigurationProperty() {
+
+        @Override
+        public Object getSource() {
+          return this;
+        }
+
+        @Override
+        public String getValue() {
+          NumberFormat nf = NumberFormat.getInstance(new Locale("pt", "PT"));
```

Comment:
get the `pt`, `PT` values from the `configurationAttributeKey` so that test cases for other locales can be added easily if needed.